src/services/depositService.ts

```diff
@@ -141,6 +140,30 @@ export function createDepositEVMERC20Transaction(
   }
 }
 
+export function createDepositSiloToSiloTransaction(
+  assetAccountId: string,
+  userAddress: string,
+  amount: bigint,
+  depositAddress: string,
+  siloAddress: string
+): { to: Address; data: Hash } {
+  // TODO check chain type for userAddress as a rule, pass through the user chainType
+  const data = encodeFunctionData({
+    abi: siloAbi,
+    functionName: "safeFtTransferCallToNear",
+    args: [
+      assetAccountId as Address,
```

Comment:
In order to have a proper type safety you can use `getAddress` from viem. It will also check validity of the address and throw if address is not correct.